### PR TITLE
8338333: Add jls links to javax.lang.model.element.Modifier

### PR DESCRIPTION
**Pull request by jddarcy**
The javadoc in javax.lang.model.element.Modifier would be more informative to readers if it linked to the relevant JLS sections discussing that modifier.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8338333](https://bugs.openjdk.org/browse/JDK-8338333): Add jls links to javax.lang.model.element.Modifier (**Enhancement** - P4)

### Reviewers
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - **Reviewer**)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**) 🔄 Re-review required (review applies to [2e45196b](https://git.openjdk.org/jdk/pull/20572/files/2e45196ba3834cd25fa967939676e5b4c4cebae9))
 * [Jan Lahoda](https://openjdk.org/census#jlahoda) (@lahodaj - **Reviewer**) 🔄 Re-review required (review applies to [2e45196b](https://git.openjdk.org/jdk/pull/20572/files/2e45196ba3834cd25fa967939676e5b4c4cebae9))
 * [Pavel Rappo](https://openjdk.org/census#prappo) (@pavelrappo - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20572/head:pull/20572` \
`$ git checkout pull/20572`

Update a local copy of the PR: \
`$ git checkout pull/20572` \
`$ git pull https://git.openjdk.org/jdk.git pull/20572/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20572`

View PR using the GUI difftool: \
`$ git pr show -t 20572`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20572.diff">https://git.openjdk.org/jdk/pull/20572.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20572#issuecomment-2287029611)

### CODE REVIEW

**Review by liach — APPROVED** [review]
The section numbers and titles (code fonts, casing) look all right.

File: src/java.compiler/share/classes/javax/lang/model/element/Modifier.java [hunk 1]
```diff
@@ -51,38 +51,119 @@ public enum Modifier {
...
+     * @jls 9.4 Method Declarations
      * @since 1.8
      */
      DEFAULT,
-    /** The modifier {@code static} */          STATIC,
+
+    /**
+     * The modifier {@code static}
+     *
+     * @jls 8.1.1.4 {@code static} Classes
```
Comment (pavelrappo) [inline src/java.compiler/share/classes/javax/lang/model/element/Modifier.java:95]: Do we need to add this?

    @jls 9.1.1.3 {@code static} Interfaces

Comment (jddarcy) [reply src/java.compiler/share/classes/javax/lang/model/element/Modifier.java:95]: Good catch @pavelrappo ; fixed.

**Review by irisclark — APPROVED** [review]
References appear correct as compared to the annotated version of the JLS version for Java SE 23:

https://cr.openjdk.org/~iris/se/23/latestSpec/java-se-23-jls-fr-diffs.pdf

**Review by vicente-romero-oracle — APPROVED** [review]
lgtm

**Review by irisclark — APPROVED** [review]

**Review by liach — APPROVED** [review]

**Pull request closed without merging**